engine/step/step.go

```diff
@@ -493,13 +501,30 @@ func AfterRun(st *Step, values *values.Values, ss StateSetter) {
 				break
 			}
 		}
-		for step, state := range sc.Then {
-			if step == stepRefThis {
-				step = st.Name
+		switch sc.Type {
+		case condition.LOOP:
+			if st.MaxRetries < 0 || st.TryCount <= st.MaxRetries {
+				state := fmt.Sprintf("%s%d", StateLooping, st.TryCount)
+				ss(st.Name, state, "")
+			} else {
+				ss(st.Name, StateDone, "")
+				logrus.Debugf("AfterRun: Step [%s] loop condition: max_retries (%d) exceeded", st.Name, st.MaxRetries)
+			}
+		case condition.CHECK:
+			for step, state := range sc.Then {
+				if step == stepRefThis {
+					step = st.Name
+				}
+				ss(step, state, sc.Message)
 			}
-			ss(step, state, sc.Message)
 		}
 	}
+
+	// check the status after the conditions’ evaluation in case a "check"
+	// condition changed the status
+	if strings.HasPrefix(st.State, StateLooping) {
+		time.Sleep(st.LoopDelay.Duration)
```

Comment:
I'm not sure to understand why this might be needed

Reply:
Ok, I saw @wI2L comment

> Set a delay between each execution of the step.

Do you have more insights about having a pause between execution ?
Having a `time.Sleep` in the main thread will prevent others steps collection, and could lead to data loss if the application receive a `SIGTERM`.